packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -252,10 +252,14 @@ class DefaultTextEditingShortcuts extends Shortcuts {
     ..._commonShortcuts,
     SingleActivator(LogicalKeyboardKey.home): ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
     SingleActivator(LogicalKeyboardKey.end): ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
-    SingleActivator(LogicalKeyboardKey.home, shift: true): ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false),
-    SingleActivator(LogicalKeyboardKey.end, shift: true): ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.home, shift: true): ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false, collapseAtReversal: true),
```

Comment:
I tried on windows 10, it doesn't seem to collapse at reversal? E.g., the initial selection is `aaaaaa|bbbbb`, I pressed <kbd>shift</kbd>-<kbd>End</kbd> it became `aaaaaa<bbbbb>` and then I pressed <kbd>shift</kbd>-<kbd>Home</kbd> it became `<aaaaaa>bbbbb`. This was tested in the explorer's address bar.

Reply:
Thanks good catch.  My table in the description of this PR is correct, but I got this code backwards when I ported it.  I've fixed it.

I also double checked Linux (gedit).  It has the same behavior as Windows.